tests/lib/devicetree/src/main.c

```diff
@@ -289,6 +289,17 @@ static void test_bus(void)
 		     "inst 0 i2c dev label");
 	zassert_true(!strncmp(i2c_bus, DT_INST_BUS_LABEL(0), strlen(i2c_bus)),
 		     "inst 0 i2c bus label");
+
+#undef DT_DRV_COMPAT
+	/*
+	 * Make sure the underlying DT_COMPAT_ON_BUS used by
+	 * DT_ANY_INST_ON_BUS works without DT_DRV_COMPAT defined.
+	 */
+	zassert_equal(DT_COMPAT_ON_BUS(vnd_spi_device, spi), 1, NULL);
+	zassert_equal(DT_COMPAT_ON_BUS(vnd_spi_device, i2c), 0, NULL);
+
+	zassert_equal(DT_COMPAT_ON_BUS(vnd_i2c_device, i2c), 1, NULL);
+	zassert_equal(DT_COMPAT_ON_BUS(vnd_i2c_device, spi), 0, NULL);
```

Comment:
can we add a test for having the same compat on both i2c and spi?

Reply:
> can we add a test for having the same compat on both i2c and spi?

sure, done

Reply:
Did I miss something here?  was expecting something like:

```
zassert_equal(DT_COMPAT_ON_BUS(foo, i2c), 1, NULL);
zassert_equal(DT_COMPAT_ON_BUS(foo, spi), 1, NULL);
```

Reply:
The push must have failed without me noticing. Pushed now.